src/main/java/frc/robot/commands/AutoPickupNote.java

```diff
@@ -0,0 +1,124 @@
+        yTranslationPidController.setSetpoint(0);
+    
+        // creating xTranslationPidController and setting the toleance and setpoint
+        xTranslationPidController = new PIDController(0, 0, 0);
+        xTranslationPidController.setSetpoint(0);
+        
+        // creating rotationPidController and setting the toleance and setpoint
+        rotationPidController = new PIDController(0, 0, 0);
+        rotationPidController.setSetpoint(0);
+
+        // puts the value of P,I and D onto the SmartDashboard
+        // Will remove later
+        SmartDashboard.putNumber("P", 0);
+        SmartDashboard.putNumber("i", 0);
+        SmartDashboard.putNumber("d", 0);
+        //THESE VALUES WILL NEED TO BE MESSED WITH, 0 FOR NOW
+
+        addRequirements(swerve, intakeSubsystem);
+    }
+    private boolean NoteSeen(){
+        return limelight.hasTag();
+    }
+    private double RadiansToDegrees(double rads){
+        return rads * (180/Math.PI);
+    }
+    private double FixedDistanceCalc(){
+        double objectHeight = 0;
+        double cameraHeight = .5;
+        double yAngle = limelight.getY();
+        double skewAngle = limelight.getSkew();
+        return (objectHeight - cameraHeight) / RadiansToDegrees(Math.tan(yAngle + skewAngle));
+    }
+
+    private double getDistanceMeters(){
+        double focalLength =  2.9272781257541 / 1000;
+        double d = (14 * .0254 * focalLength)/limelight.getArea();
+        return d;
+    }
+
+    @Override
+    public void execute(){
+        yaw = swerve.getGyroYaw();
+        
+        double xValue = limelight.getX(); //gets the limelight X Coordinate
+        double areaValue = limelight.getArea(); // gets the area percentage from the limelight
+        double angularValue = limelight.getSkew(); 
+        double distance = getDistanceMeters();
+        double fixedDistance = FixedDistanceCalc();
+
+        kP = SmartDashboard.getNumber("AlignP", 0);
+        kI = SmartDashboard.getNumber("AlignI", 0);
+        kD = SmartDashboard.getNumber("AlignD", 0);
+        
+        SmartDashboard.putNumber("Xvalue", xValue);
+        SmartDashboard.putNumber("Areavalue", areaValue);
+        SmartDashboard.putNumber("AngularValue", angularValue);
+        SmartDashboard.putNumber("Ts0", limelight.getSkew0());
+        SmartDashboard.putNumber("Ts1", limelight.getSkew1());
+        SmartDashboard.putNumber("Ts2", limelight.getSkew2());
+
+        
+        if (debouncer.calculate(NoteSeen())){
+            System.out.println("Note in view");
+            // Calculates the x and y speed values for the translation movement
+            double ySpeed = yTranslationPidController.calculate(distance * RadiansToDegrees(Math.cos(angularValue)));
+            double xSpeed = xTranslationPidController.calculate(distance * RadiansToDegrees(Math.sin(angularValue)));
+            double angularSpeed = autoUtil.calculateRotationSpeed() * Constants.Swerve.maxAngularVelocity;
```

Comment:
We need to be setting or updating the target angle for autoUtil. We also need to initialize it somewhere in this class, otherwise this will throw a NullPointerException.

Reply:
initialized and updated angle